### PR TITLE
More progress removing prototype extensions

### PR DESCRIPTION
**Pull request by RobbieTheWagner**

### CODE REVIEW

**Comment by patricklx** [comment]
it looks like computed.x are incompatible with tracked array. e.g. computed.and in `controllers/promise-tree.js`

**Comment by RobbieTheWagner** [comment]
> it looks like computed.x are incompatible with tracked array. e.g. computed.and in `controllers/promise-tree.js`.

@patricklx I am not sure I understand. What is the fix here? I do not see tracked arrays in that controller.

**Comment by patricklx** [comment]
> > it looks like computed.x are incompatible with tracked array. e.g. computed.and in `controllers/promise-tree.js`.
> 
> @patricklx I am not sure I understand. What is the fix here? I do not see tracked arrays in that controller.

The model is a tracked array. The fix is to use getters instead of computed

**Comment by RobbieTheWagner** [comment]
@patricklx I updated, but it's got a new error `Global error: Uncaught Error: Assertion Failed: EmberObject.create no longer supports defining computed properties. Define computed properties using extend() or reopen() before calling create()`

**Comment by patricklx** [comment]
computed.filter should also be replaced with getter.
I think the issue is in models/promise . There is a custom decorator which returns a computed

**Comment by RobbieTheWagner** [comment]
@patricklx do you mean `dateComputed`? Do you know how we would fix it?

File: app/controllers/promise-tree.js [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
   // below used to show the "refresh" message
-  isEmpty: equal('model.length', 0),
-  wasCleared: bool('createdAfter'),
-  neverCleared: not('wasCleared'),
-  shouldRefresh: and('isEmpty', 'neverCleared'),
+  get isEmpty() {
+    return this.model.length === 0;
+  }
+  get wasCleared() {
+    return !this.createdAfter;
```
Comment (patricklx) [inline app/controllers/promise-tree.js:27]: Should be 2 `!!`, or Boolean()

Comment (RobbieTheWagner) [inline app/controllers/promise-tree.js:27]:
```suggestion
    return Boolean(this.createdAfter);
```

Comment (RobbieTheWagner) [reply app/controllers/promise-tree.js]: Oops, good catch!

**Comment by RobbieTheWagner** [comment]
@patricklx do you think this is good to merge now or did you see anything else we needed to tweak?

**Comment by patricklx** [comment]
> @patricklx do you think this is good to merge now or did you see anything else we needed to tweak?

there are still some errors like `assembler.find(...).get is not a function`, not sure if its releated to prototype extension

**Comment by RobbieTheWagner** [comment]
@patricklx yeah, this PR is not meant to fix all errors, just a stepping stone.

**Pull request merged**